src/main.rs

```diff
@@ -189,35 +220,47 @@ fn main() -> std::io::Result<()> {
         .unicode_error_messages(!matches.is_present("NO_UNICODE"))
         .allows_charset(!matches.is_present("NO_CHARSET"));
 
-    let (mut stdout_write, mut file_write);
-    let buf_out: &mut dyn Write = if let Some(path) = matches.value_of("OUTPUT") {
-        file_write = BufWriter::new(File::open(path).unwrap_or(File::create(path)?));
-        &mut file_write
-    } else {
-        stdout_write = BufWriter::new(stdout());
-        &mut stdout_write
-    };
-
-    buf_out.write_all(
-        if let Some(name) = matches.value_of("INPUT") {
-            from_path(name, &options)
-        } else if matches.is_present("STDIN") {
-            from_string(
-                {
-                    let mut buffer = String::new();
-                    stdin().read_to_string(&mut buffer)?;
-                    buffer
-                },
-                options,
-            )
+    if matches.value_of("INPUT") == Some("-") || matches.is_present("STDIN") {
+        let mut buffer = String::new();
+        stdin().read_to_string(&mut buffer)?;
+        write_file(from_string(buffer, options), matches.value_of("OUTPUT"))
+    } else if let Some(input) = matches.value_of("INPUT") {
+        if ["sass", "scss"].contains(&input.rsplitn(2, '.').next().unwrap()) {
```

Comment:
I'd prefer to just check if it is a file or directory here. The cost of a single syscall would be offset by even a millisecond improvement in selector extension.